xarray/plot/facetgrid.py

```diff
@@ -289,13 +290,19 @@ def map_dataarray_line(self, x=None, y=None, hue=None, **kwargs):
             # None is the sentinel value
             if d is not None:
                 subset = self.data.loc[d]
-                mappable = line(subset, x=x, y=y, hue=hue,
+                mappable = plotfunc(subset, x=x, y=y, hue=hue,
                                 ax=ax, _labels=False,
                                 **kwargs)
                 self._mappables.append(mappable)
-        _, _, hueplt, xlabel, ylabel, huelabel = _infer_line_data(
-            darray=self.data.loc[self.name_dicts.flat[0]],
-            x=x, y=y, hue=hue)
+
+        if plotfunc == line:
+            _, _, hueplt, xlabel, ylabel, huelabel = _infer_line_data(
+                darray=self.data.loc[self.name_dicts.flat[0]],
+                x=x, y=y, hue=hue)
+        elif plotfunc == dataset_scatter:
```

Comment:
I think this bit should be in a separate `map_dataset` function that can be reused as the Dataset plotting API becomes more complete.

Reply:
That's easy to do. thanks.